griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -123,7 +123,10 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
 
         messages = self._prompt_stack_to_messages(prompt_stack)
 
-        params["max_tokens"] = self.max_output_tokens(self.prompt_stack_to_string(prompt_stack))
+        if isinstance(self.tokenizer, OpenAiTokenizer):
```

Comment:
Small suggestion - add a helper method to reduce duplication:

```py
def _tokens(self, prompt_stack: PromptStack) -> str | list:
    if isinstance(self.tokenizer, OpenAiTokenizer):
        return self._prompt_stack_to_messages(prompt_stack)
    else:
        return self.prompt_stack_to_string(prompt_stack)
```

Then you can use this helper to set `params["max_tokens"]` here and to return token count in the `token_count` method without adding branching in two places.